app/components/product-destroy/template.hbs

```diff
@@ -0,0 +1,8 @@
+<button data-test-product-destroy-button {{action "launchConfirmDialog"}}>{{yield}}</button>
+{{#if confirmShown}}
+  <div class="confirm-dialog">
+    <div data-test-product-destroy-text>This will remove the product, do you wish to proceed?</div>
+    <button data-test-product-destroy-confirm-button class="confirm-submit" {{action "submitConfirm"}}>OK</button>
```

Comment:
As the component will be more general one we should change

* `data-test-warning-msg`
* `data-test-confirm-button`
* `data-test-cancel-button`